packages/fixie/src/agent.ts

```diff
@@ -260,28 +260,31 @@ export class FixieAgent {
   /** Package the code in the given directory and return the path to the tarball. */
   private static getCodePackage(agentPath: string): string {
     // Read the package.json file to get the package name and version.
-
     const packageJsonPath = path.resolve(path.join(agentPath, 'package.json'));
     const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, 'utf8'));
 
     // Create a temporary directory and run `npm pack` inside.
     const tempdir = fs.mkdtempSync(path.join(os.tmpdir(), `fixie-tmp-${packageJson.name}-${packageJson.version}-`));
-    const commandline = `npm pack ${path.resolve(agentPath)} --silent >/dev/null`;
-    execSync(commandline, { cwd: tempdir });
+    const commandline = `npm pack ${path.resolve(agentPath)}`;
+    try {
+      execSync(commandline, { cwd: tempdir, stdio: 'inherit' });
+    } catch (ex) {
+      throw new Error(`\`${commandline}\` failed. Check for build errors above and retry.`);
+    }
     return `${tempdir}/${packageJson.name}-${packageJson.version}.tgz`;
   }
 
   /** Create a new agent revision, which deploys the agent. */
-  private async createRevision({
-    externalUrl,
-    tarball,
-    environmentVariables,
-  }: {
-    externalUrl?: string;
-    tarball?: string;
-    environmentVariables: Record<string, string>;
-  }): Promise<AgentRevision> {
-    const uploadFile = tarball ? fs.readFileSync(fs.realpathSync(tarball)) : undefined;
+  private async createRevision(
+    opts:
+      | { externalUrl: string; tarball?: undefined }
+      | {
+          externalUrl?: undefined;
+          tarball: string;
```

Comment:
It looks like you want `MergeExclusive` here from type-fest.

Reply:
Nice!